src/OmniSharp.Roslyn.CSharp/CSharpLanguage.cs

```diff
@@ -0,0 +1,15 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using OmniSharp.Mef;
+
+namespace OmniSharp.Roslyn.CSharp.Services
+{
+    public class LanguagePredicate
+    {
+        private static readonly string[] ValidCSharpExtensions = { "cs", "csx", "cake" };
+        [OmniSharpLanguage(LanguageNames.CSharp)]
+        public Func<string, Task<bool>> IsApplicableTo { get; } = filePath => Task.FromResult(ValidCSharpExtensions.Any(extension => filePath.EndsWith(extension)));
```

Comment:
If the plugin is out of process, then we might be waiting for stdio to give us a result, figured this would make that logic easier.  Though thinking about it, maybe it won't be needed.